railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -243,6 +249,17 @@ def config_target_version
     end
   end
 
+  class MasterGemfileGenerator < AppBuilder # :nodoc:
+    def install_shell_gemfile
+      template "GemfileMasterRailsInstaller", "Gemfile"
```

Comment:
So we're using this to essentially build a local Rails checkout that can then generate an application?

Not really liking installing this Gemfile that'll linger afterward. We may as well just do a shallow clone via `git clone https://github.com/rails/rails.git`. We essentially don't need any other lines in the file and if we install the checkout at a known location, we can do `/path/to/rails/railties/exe/rails new …`